src/machines/activity_state_machine.rs

```diff
@@ -42,17 +58,155 @@ fsm! {
+            shared_state: SharedState {
+                attrs: attribs,
+                cancellation_type: ActivityCancellationType::TryCancel,
+            },
+        };
+        s.on_event_mut(ActivityMachineEvents::Schedule)
+            .expect("Scheduling activities doesn't fail");
+        let cmd = Command {
+            command_type: CommandType::ScheduleActivityTask as i32,
+            attributes: Some(s.shared_state().attrs.clone().into()),
+        };
+        (s, cmd)
+    }
+}
+
+impl TryFrom<HistoryEvent> for ActivityMachineEvents {
+    type Error = WFMachinesError;
+
+    fn try_from(e: HistoryEvent) -> Result<Self, Self::Error> {
+        Ok(match EventType::from_i32(e.event_type) {
+            Some(EventType::ActivityTaskScheduled) => Self::ActivityTaskScheduled,
+            Some(EventType::ActivityTaskStarted) => Self::ActivityTaskStarted,
+            Some(EventType::ActivityTaskCompleted) => {
+                if let Some(history_event::Attributes::ActivityTaskCompletedEventAttributes(
+                    attrs,
+                )) = e.attributes
+                {
+                    Self::ActivityTaskCompleted(attrs)
+                } else {
+                    return Err(WFMachinesError::MalformedEvent(
+                        e,
+                        "Activity completion attributes were unset".to_string(),
+                    ));
+                }
+            }
+            Some(EventType::ActivityTaskFailed) => Self::ActivityTaskFailed,
+            Some(EventType::ActivityTaskTimedOut) => Self::ActivityTaskTimedOut,
+            Some(EventType::ActivityTaskCancelRequested) => Self::ActivityTaskCancelRequested,
+            Some(EventType::ActivityTaskCanceled) => Self::ActivityTaskCanceled,
+            _ => {
+                return Err(WFMachinesError::UnexpectedEvent(
+                    e,
+                    "Activity machine does not handle this event",
+                ))
+            }
+        })
+    }
+}
+
+impl TryFrom<CommandType> for ActivityMachineEvents {
+    type Error = ();
+
+    fn try_from(c: CommandType) -> Result<Self, Self::Error> {
+        Ok(match c {
+            CommandType::ScheduleActivityTask => Self::CommandScheduleActivityTask,
+            CommandType::RequestCancelActivityTask => Self::CommandRequestCancelActivityTask,
+            _ => return Err(()),
+        })
+    }
+}
+
+impl WFMachinesAdapter for ActivityMachine {
+    fn adapt_response(
+        &self,
+        event: &HistoryEvent,
+        has_next_event: bool,
+        my_command: ActivityMachineCommand,
+    ) -> Result<Vec<MachineResponse>, WFMachinesError> {
+        Ok(match my_command {
+            ActivityMachineCommand::Complete(result) => vec![ResolveActivity {
+                activity_id: self.shared_state.attrs.activity_id.clone(),
+                result: Some(ActivityResult {
+                    status: Some(activity_result::Status::Completed(ActivityTaskSuccess {
```

Comment:
I assume we're intentionally only handling success case right now?

Reply:
Yes, I'll add failure and cancelation soon as well.